doc/protocols/scion-header.rst

```diff
@@ -887,25 +898,15 @@ CurrHF
 HFCount
     The number of total HopFields.
 Reservation ID Suffix
-    Uses 12 bytes. Either an E2E Reservation ID suffix or a
+    Uses 10 bytes. Either an E2E Reservation ID suffix or a
     Segment Reservation ID suffix,
     depending on `S`. If :math:`S=1`, the Segment Reservation ID suffix
-    is padded with zeroes until using all 12 bytes. If :math:`S=0`
-    the 12 bytes from the E2E Reservation ID suffix are included.
-Expiration Tick
-    The value represents the "tick" where this packet is no longer valid.
-    A tick is four seconds, so :math:`\text{Expiration Time} = 4 \times
-    \text{Expiration Tick}` seconds after Unix epoch.
+    is padded with zeroes until using all 10 bytes. If :math:`S=0`
+    the 10 bytes from the E2E Reservation ID suffix are included.
```

Comment:
IIRC, for EER's, the reservation suffix was chosen to be 12 bytes, so that it can be sampled at random with a very low collision probability.
Are you planning to deviate from this?
- If yes, how do you plan to keep track of which EER suffixes are still in use, and which ones can be assigned to new reservations?
- If not, what is the collision probability when using 10 bytes instead of 12?

Reply:
Update:
As discussed, the plan was to choose the suffix for EERs at random.
However, if the Colibri service itself creates those suffixes and not the end hosts, is random sampling still necessary?
-> If an AS just increases a counter to generate a new suffix, then 6 bytes should be enough already: even with 100'000 new reservations per second, it would take 89 years until the counter overflows, no reservation will stay valid that long. With 4 bytes it would take ~12 hours until the counter overflows.

If the suffixes should still be sampled at random, then I think 8 bytes could be enough:
With 10^7 hosts in an AS, each having 100 reservations, the probability of a collision is 2.67%.
In the rare case of a collision, the AS can simply sample a new suffix.
https://kevingal.com/apps/collision.html